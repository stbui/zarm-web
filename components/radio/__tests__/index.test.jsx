import React from 'react';
import { render, mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import Radio from '../index';

describe('Radio', () => {
  it('renders normal Radio correctly', () => {
    const wrapper = render(
      <Radio value="a">选择</Radio>,
    );

    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('renders disabled Radio correctly', () => {
    const wrapper = render(
      <Radio disabled value="a">
        选择
      </Radio>,
    );

    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('renders checked Radio correctly', () => {
    const wrapper = render(
      <Radio checked>选择</Radio>,
    );

    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('behaves correctly when changing radio status', () => {
    const onChange = jest.fn();
    const wrapper = mount(
      <Radio value="a" onChange={onChange}>
        选择
      </Radio>,
    );

    wrapper.find('.zw-radio__input').simulate('change');
    expect(onChange).toBeCalled();
  });

  it('renders Radio Group correctly', () => {
    const wrapper = render(
      <Radio.Group>
        <Radio value="a">A</Radio>
        <Radio value="b">B</Radio>
        <Radio value="c">C</Radio>
        <Radio value="d">D</Radio>
      </Radio.Group>,
    );

    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('renders defaultValue Radio Group correctly', () => {
    const wrapper = render(
      <Radio.Group defaultValue="a">
        <Radio value="a">A</Radio>
        <Radio value="b">B</Radio>
        <Radio value="c">C</Radio>
        <Radio value="d">D</Radio>
      </Radio.Group>,
    );

    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('renders value Radio Group correctly', () => {
    const wrapper = render(
      <Radio.Group value="a">
        <Radio value="a">A</Radio>
        <Radio value="b">B</Radio>
        <Radio value="c">C</Radio>
        <Radio value="d">D</Radio>
      </Radio.Group>,
    );

    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('renders checked Radio Group correctly', () => {
    const wrapper = render(
      <Radio.Group>
        <Radio value="a">A</Radio>
        <Radio value="b" checked>B</Radio>
        <Radio value="c">C</Radio>
        <Radio value="d">D</Radio>
      </Radio.Group>,
    );

    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('behaves correctly when choosing from Radio Group', () => {
    const onChange = jest.fn();
    const wrapper = mount(
      <Radio.Group onChange={onChange}>
        <Radio value="a">A</Radio>
        <Radio value="b">B</Radio>
        <Radio value="c">C</Radio>
        <Radio value="d">D</Radio>
      </Radio.Group>,
    );

    wrapper
      .find('.zw-radio__input')
      .first()
      .simulate('change');
    expect(onChange).toBeCalled();
  });
});
