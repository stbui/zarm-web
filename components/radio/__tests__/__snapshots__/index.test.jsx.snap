// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Radio renders Radio Group correctly 1`] = `
<div
  class="zw-radio-group zw-radio-group--md zw-radio-group--radius zw-radio-group--compact zw-radio-group--ghost"
>
  <div
    class="zw-radio-group__inner"
  >
    <span
      class="zw-radio"
    >
      <span
        class="zw-radio__widget"
      >
        <span
          class="zw-radio__inner"
        />
      </span>
      <span
        class="zw-radio__text"
      >
        A
      </span>
      <input
        class="zw-radio__input"
        type="radio"
        value="a"
      />
    </span>
    <span
      class="zw-radio"
    >
      <span
        class="zw-radio__widget"
      >
        <span
          class="zw-radio__inner"
        />
      </span>
      <span
        class="zw-radio__text"
      >
        B
      </span>
      <input
        class="zw-radio__input"
        type="radio"
        value="b"
      />
    </span>
    <span
      class="zw-radio"
    >
      <span
        class="zw-radio__widget"
      >
        <span
          class="zw-radio__inner"
        />
      </span>
      <span
        class="zw-radio__text"
      >
        C
      </span>
      <input
        class="zw-radio__input"
        type="radio"
        value="c"
      />
    </span>
    <span
      class="zw-radio"
    >
      <span
        class="zw-radio__widget"
      >
        <span
          class="zw-radio__inner"
        />
      </span>
      <span
        class="zw-radio__text"
      >
        D
      </span>
      <input
        class="zw-radio__input"
        type="radio"
        value="d"
      />
    </span>
  </div>
</div>
`;

exports[`Radio renders checked Radio Group correctly 1`] = `
<div
  class="zw-radio-group zw-radio-group--md zw-radio-group--radius zw-radio-group--compact zw-radio-group--ghost"
>
  <div
    class="zw-radio-group__inner"
  >
    <span
      class="zw-radio"
    >
      <span
        class="zw-radio__widget"
      >
        <span
          class="zw-radio__inner"
        />
      </span>
      <span
        class="zw-radio__text"
      >
        A
      </span>
      <input
        class="zw-radio__input"
        type="radio"
        value="a"
      />
    </span>
    <span
      class="zw-radio zw-radio--checked"
    >
      <span
        class="zw-radio__widget"
      >
        <span
          class="zw-radio__inner"
        />
      </span>
      <span
        class="zw-radio__text"
      >
        B
      </span>
      <input
        checked=""
        class="zw-radio__input"
        type="radio"
        value="b"
      />
    </span>
    <span
      class="zw-radio"
    >
      <span
        class="zw-radio__widget"
      >
        <span
          class="zw-radio__inner"
        />
      </span>
      <span
        class="zw-radio__text"
      >
        C
      </span>
      <input
        class="zw-radio__input"
        type="radio"
        value="c"
      />
    </span>
    <span
      class="zw-radio"
    >
      <span
        class="zw-radio__widget"
      >
        <span
          class="zw-radio__inner"
        />
      </span>
      <span
        class="zw-radio__text"
      >
        D
      </span>
      <input
        class="zw-radio__input"
        type="radio"
        value="d"
      />
    </span>
  </div>
</div>
`;

exports[`Radio renders checked Radio correctly 1`] = `
<span
  class="zw-radio zw-radio--checked"
>
  <span
    class="zw-radio__widget"
  >
    <span
      class="zw-radio__inner"
    />
  </span>
  <span
    class="zw-radio__text"
  >
    选择
  </span>
  <input
    checked=""
    class="zw-radio__input"
    type="radio"
  />
</span>
`;

exports[`Radio renders defaultValue Radio Group correctly 1`] = `
<div
  class="zw-radio-group zw-radio-group--md zw-radio-group--radius zw-radio-group--compact zw-radio-group--ghost"
>
  <div
    class="zw-radio-group__inner"
  >
    <span
      class="zw-radio zw-radio--checked"
    >
      <span
        class="zw-radio__widget"
      >
        <span
          class="zw-radio__inner"
        />
      </span>
      <span
        class="zw-radio__text"
      >
        A
      </span>
      <input
        checked=""
        class="zw-radio__input"
        type="radio"
        value="a"
      />
    </span>
    <span
      class="zw-radio"
    >
      <span
        class="zw-radio__widget"
      >
        <span
          class="zw-radio__inner"
        />
      </span>
      <span
        class="zw-radio__text"
      >
        B
      </span>
      <input
        class="zw-radio__input"
        type="radio"
        value="b"
      />
    </span>
    <span
      class="zw-radio"
    >
      <span
        class="zw-radio__widget"
      >
        <span
          class="zw-radio__inner"
        />
      </span>
      <span
        class="zw-radio__text"
      >
        C
      </span>
      <input
        class="zw-radio__input"
        type="radio"
        value="c"
      />
    </span>
    <span
      class="zw-radio"
    >
      <span
        class="zw-radio__widget"
      >
        <span
          class="zw-radio__inner"
        />
      </span>
      <span
        class="zw-radio__text"
      >
        D
      </span>
      <input
        class="zw-radio__input"
        type="radio"
        value="d"
      />
    </span>
  </div>
</div>
`;

exports[`Radio renders disabled Radio correctly 1`] = `
<span
  class="zw-radio zw-radio--disabled"
>
  <span
    class="zw-radio__widget"
  >
    <span
      class="zw-radio__inner"
    />
  </span>
  <span
    class="zw-radio__text"
  >
    选择
  </span>
  <input
    class="zw-radio__input"
    disabled=""
    type="radio"
    value="a"
  />
</span>
`;

exports[`Radio renders normal Radio correctly 1`] = `
<span
  class="zw-radio"
>
  <span
    class="zw-radio__widget"
  >
    <span
      class="zw-radio__inner"
    />
  </span>
  <span
    class="zw-radio__text"
  >
    选择
  </span>
  <input
    class="zw-radio__input"
    type="radio"
    value="a"
  />
</span>
`;

exports[`Radio renders value Radio Group correctly 1`] = `
<div
  class="zw-radio-group zw-radio-group--md zw-radio-group--radius zw-radio-group--compact zw-radio-group--ghost"
>
  <div
    class="zw-radio-group__inner"
  >
    <span
      class="zw-radio zw-radio--checked"
    >
      <span
        class="zw-radio__widget"
      >
        <span
          class="zw-radio__inner"
        />
      </span>
      <span
        class="zw-radio__text"
      >
        A
      </span>
      <input
        checked=""
        class="zw-radio__input"
        type="radio"
        value="a"
      />
    </span>
    <span
      class="zw-radio"
    >
      <span
        class="zw-radio__widget"
      >
        <span
          class="zw-radio__inner"
        />
      </span>
      <span
        class="zw-radio__text"
      >
        B
      </span>
      <input
        class="zw-radio__input"
        type="radio"
        value="b"
      />
    </span>
    <span
      class="zw-radio"
    >
      <span
        class="zw-radio__widget"
      >
        <span
          class="zw-radio__inner"
        />
      </span>
      <span
        class="zw-radio__text"
      >
        C
      </span>
      <input
        class="zw-radio__input"
        type="radio"
        value="c"
      />
    </span>
    <span
      class="zw-radio"
    >
      <span
        class="zw-radio__widget"
      >
        <span
          class="zw-radio__inner"
        />
      </span>
      <span
        class="zw-radio__text"
      >
        D
      </span>
      <input
        class="zw-radio__input"
        type="radio"
        value="d"
      />
    </span>
  </div>
</div>
`;
